scipy/sparse/base.py

```diff
@@ -128,17 +128,35 @@ def reshape(self, shape, order='C'):
         raise NotImplementedError("Reshaping not implemented for %s." %
                                   self.__class__.__name__)
 
-    def astype(self, t):
+    def astype(self, dtype, casting='unsafe', copy=True):
         """Cast the matrix elements to a specified type.
 
         The data will be copied.
```

Comment:
Unless `copy=False` and dtypes match. Or just remove this line, as it's explained below.